sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -122,57 +181,95 @@ bool test_main(void) {
   CHECK_DIF_OK(dif_rv_core_ibex_init(
       mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR), &ibex));
 
-  dif_rstmgr_reset_info_bitfield_t rst_info;
-  rst_info = rstmgr_testutils_reason_get();
+  switch (rstmgr_testutils_reason_get()) {
+    case kDifRstmgrResetInfoPor:  // The first power-up.
+      LOG_INFO("Triggering single fault.");
 
-  if (rst_info == kDifRstmgrResetInfoPor) {
-    LOG_INFO("Booting for the first time, setting wdog");
+      // Enable cpu info.
+      CHECK_DIF_OK(dif_rstmgr_cpu_info_set_enabled(&rstmgr, kDifToggleEnabled));
 
-    uint32_t bark_cycles = aon_timer_testutils_get_aon_cycles_from_us(100);
-    uint32_t bite_cycles = aon_timer_testutils_get_aon_cycles_from_us(100);
+      double_fault = false;
+      OT_ADDRESSABLE_LABEL(kSingleFaultAddrLower);
+      addr_val = *kIllegalAddr0;
+      OT_ADDRESSABLE_LABEL(kSingleFaultAddrUpper);
+      LOG_ERROR(
+          "This should be unreachable; a single fault should have occured.");
```

Comment:
You'd better using `CHECK(false,"This should be unreachable; a single fault should have occured.")` here to make sure the test will not procede.